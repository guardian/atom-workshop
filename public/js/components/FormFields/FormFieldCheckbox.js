import React, {PropTypes} from 'react';
import ShowErrors from '../Utilities/ShowErrors';
import { errorPropType } from '../../constants/errorPropType';

export default class FormFieldCheckbox extends React.Component {

  static propTypes = {
    fieldLabel: PropTypes.string,
    fieldName: PropTypes.string.isRequired,
    fieldValue: PropTypes.bool.isRequired,
    fieldErrors: PropTypes.arrayOf(errorPropType),
    onUpdateField: PropTypes.func.isRequired
  }

  onUpdate = (e) => {
    this.props.onUpdateField(e.target.value);
  }

  render() {
    return (
      <div className="form__group form__group--checkbox">
        {this.props.fieldLabel ? <label className="form__label" htmlFor={this.props.fieldName}>{this.props.fieldLabel}</label> : false}
        <input className="form__checkbox" type="checkbox" checked={this.props.fieldValue} name={this.props.fieldName} value={this.props.fieldValue} onChange={this.onUpdate} />
        {!this.props.fieldLabel ? <span className="form__label form__label--checkbox">{this.props.fieldName}</span> : false}
        <ShowErrors errors={this.props.fieldErrors}/>
      </div>
    );
  }
}
