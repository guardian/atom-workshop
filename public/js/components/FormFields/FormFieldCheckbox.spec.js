import React from 'react';
import FormFieldCheckbox from './FormFieldCheckbox';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

let fieldLabel = 'test',
    fieldName = 'test',
    fieldValue = true;

test('Should render', () => {

  const updateFn = jest.fn();
  const component = renderer.create(
    <FormFieldCheckbox fieldLabel={fieldLabel} fieldName={fieldName} fieldValue={fieldValue} onUpdateField={updateFn} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should call update function on change', () => {

  const updateFn = jest.fn();
  const checkbox = shallow(
    <FormFieldCheckbox fieldLabel={fieldLabel} fieldName={fieldName} fieldValue={fieldValue} onUpdateField={updateFn} />
  );

  checkbox.find('input').simulate('change', {target: {value: false}});

  expect(updateFn).toHaveBeenCalledTimes(1);
});
