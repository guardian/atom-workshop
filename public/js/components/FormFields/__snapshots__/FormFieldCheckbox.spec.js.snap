exports[`test Should render 1`] = `
<div
  className="form__group form__group--checkbox">
  <label
    className="form__label"
    htmlFor="test">
    test
  </label>
  <input
    checked={true}
    className="form__checkbox"
    name="test"
    onChange={[Function]}
    type="checkbox"
    value={true} />
  <div />
</div>
`;
